test/new-e2e/runner/configmap.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/runner/parameters"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+)
+
+type ConfigMap auto.ConfigMap
+
+func (cm ConfigMap) Set(key, val string, secret bool) {
+	cm[key] = auto.ConfigValue{
+		Value:  val,
+		Secret: secret,
+	}
+}
+
+func (cm ConfigMap) Merge(in ConfigMap) {
+	for key, val := range in {
+		cm[key] = val
+	}
+}
+
+func (cm ConfigMap) ToPulumi() auto.ConfigMap {
+	return (auto.ConfigMap)(cm)
+}
+
+func SetConfigMapFromParameter(store parameters.Store, cm ConfigMap, paramName, configMapKey string) error {
```

Comment:
It s getting parameters from the store, it knows about store.

Reply:
If you don't want `ConfigMap` to know about `Store`, you need to pass a dictionary instead of a store

Reply:
No the `ConfigMap` object does not know about the `Store`. However, a free function in the same package (for convenience, but we could put it elsewhere) that uses the public API of both objects to do its job, knows about the Store, which I believe, is quite different.

Reply:
What I meant is that this function knows that is is using a `Store` to update a `ConfigMap`, so it could be called `SetConfigMapFromStoreParameter` instead of `SetConfigMapFromParameter`